infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/infrastructure/CorfuReplicationDiscoveryService.java

```diff
@@ -484,11 +485,9 @@ public void processLockRelease() {
         log.debug("Lock released");
         // Unset isLeader flag after stopping log replication
         stopLogReplication();
+        interClusterServerNode.close();
```

Comment:
any reason we now invoke close() on losing the leadership?

Reply:
interClusterServerNode is started only on the leader node.
If a leader node looses leadership, it also shuts down the interClusterServerNode. This is a small optimization and think this way maybe helpful in sharding as well

But I realized I was missing a null check here as interClusterServerNode can be null. Adding it.